libs/community/langchain_community/tools/audio/huggingface_text_to_speech_inference.py

```diff
@@ -0,0 +1,100 @@
+import os
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+
+import requests
+from langchain_core.callbacks import CallbackManagerForToolRun
+from langchain_core.pydantic_v1 import SecretStr
+from langchain_core.tools import BaseTool
+
+
+class HuggingFaceSupportedAudioFormat(Enum):
+    WAV = WAVE = WAVEFORM = "wav"
+
+
+class HuggingFaceTextToSpeechModelInference(BaseTool):
+    """
+    HuggingFace Text-to-Speech Model Inference.
+
+    Requirements:
+
+        - Environment variable ``HUGGINGFACE_API_KEY`` must be set,
+          or passed to the constructor.
+    """
+
+    name: str = "openai_text_to_speech"
+    description: str = "A wrapper around OpenAI Text-to-Speech API. "
+
+    model: str
+    api_url: str
+    huggingface_api_key: SecretStr
+    format: HuggingFaceSupportedAudioFormat
+    output_dir: str
+
+    _DEFAULT_OUTPUT_DIR = "tts_output"
+    _DEFAULT_OUTPUT_NAME = "output"
+    _HUGGINGFACE_API_KEY_ENV_NAME = "HUGGINGFACE_API_KEY"
+    _HUGGINGFACE_API_URL_ROOT = "https://api-inference.huggingface.co/models"
+
+    def __init__(
+        self,
+        model: str,
+        format: HuggingFaceSupportedAudioFormat,
+        huggingface_api_key: Optional[SecretStr] = None,
+        output_dir: Optional[str] = None,
+    ) -> None:
+        if not huggingface_api_key:
+            huggingface_api_key = SecretStr(
+                os.getenv(self._HUGGINGFACE_API_KEY_ENV_NAME, "")
+            )
+
+        if (
+            not huggingface_api_key
+            or not huggingface_api_key.get_secret_value()
+            or huggingface_api_key.get_secret_value() == ""
+        ):
+            raise ValueError(
+                f"'{self._HUGGINGFACE_API_KEY_ENV_NAME}' must be or set or passed"
+            )
+
+        super().__init__(
+            model=model,
+            format=format,
+            api_url=f"{self._HUGGINGFACE_API_URL_ROOT}/{model}",
+            huggingface_api_key=huggingface_api_key,
+            output_dir=output_dir if output_dir else self._DEFAULT_OUTPUT_DIR,
+        )
+
+    def _run(
+        self,
+        query: str,
+        output_name: Optional[str] = None,
+        run_manager: Optional[CallbackManagerForToolRun] = None,
+    ) -> str:
+        response = requests.post(
+            self.api_url,
+            headers={
+                "Authorization": f"Bearer {self.huggingface_api_key.get_secret_value()}"
+            },
+            json={"inputs": query},
+        )
+
+        audio_bytes = response.content
+
+        output_name = (
+            f"{output_name or self._default_output_name()}.{self.format.value}"
+        )
+        output_path = os.path.join(self.output_dir, output_name)
```

Comment:
This logic forces the user to reinstantiate the tool every time. I think it would be better to have an option to generate an output_name that's a uuid.uuid4. what do you think about making that the default? Users would be allowed to over-ride to specify a name if they really want

Reply:
I thought it would be good for the user to instantiate the tool with a directory, so it doesn't give flood the directory where its called from, and then output everything there, this way they can define either a tmp directory they will throw out or a place they want to save it. 

I am using a timestamp rather than a UUID if the user does not specify a specific filename. I will switch it to a UUID though.

Reply:
But yeah I will remove the output_dir field and the user can handle that logic by optionally passing in a file name.